addons/barcodes/static/src/components/manual_barcode.js

```diff
@@ -0,0 +1,42 @@
+import { BarcodeDialog } from '@web/core/barcode/barcode_dialog';
+import { Component, onMounted, useRef, useState } from "@odoo/owl";
+
+export class BarcodeInput extends Component {
+    static template = "barcodes.BarcodeInput";
+    static props = {
+        onSubmit: Function,
+        placeholder: { type: String, optional: true },
+    };
+
+    setup() {
+        this.state = useState({
+            barcode: false,
+        });
+        this.barcodeManual = useRef('manualBarcode');
+        // Autofocus processing was blocked because a document already has a focused element.
+        onMounted(() => {
+            this.barcodeManual.el.focus();
+        });
+    }
+
+    /**
+     * Called when press Enter after filling barcode input manually.
+     *
+     * @private
+     * @param {KeyboardEvent} ev
+     */
+    _onKeydown(ev) {
+        if (ev.key === "Enter" && this.state.barcode) {
+            this.props.onSubmit(this.state.barcode);
+        }
+    }
```

Comment:
Instead of doing that, we can use `useHotkey` hook. Either we can use `getActiveHotkey` from the hotkey service.